src/client/views/select-wallet.ts

```diff
@@ -274,8 +274,12 @@ export class SelectWallet extends AbstractView {
 		)
 
 		try {
-			await this.client.negotiatorConnectToWallet(wallet)
+			let walletAddress = await this.client.negotiatorConnectToWallet(wallet)
 			this.ui.dismissLoader()
+			
+			// if user rejected connect to account
+			if (!walletAddress) throw new Error("Wallet didn't connect")
```

Comment:
If the exception propagates, this shouldn't be necessary.

Reply:
@oleggrib Good point but this can be thrown in Web3WalletProvider class so it works for all UIs (i.e. BC). 
Maybe at the end of Web3WalletProvider.connectWith() method?